### ISSUE TITLE
[Windows] User Code Snippets not working + code formatting wrong

### ISSUE THREAD

**Issue by Grisgram** [YoYoGames/GameMaker-Bugs#4994]
### Description
Hey there,

### Part 1
If I am right, then user code-snippets should be in
`C:\ProgramData\GameMakerStudio2\User\snippets.txt`

I created the file `snippets.txt`, containing only  one snippet to test
```
For-Array:array_foreach(|array|), function(_item, _idx) {$ $}
```
The snippet does not show up (after restarting GM) - no user snippets show up, no matter how many lines I add.

So I did, what one should _not_ do: I edited the "original" snippets.txt in `C:\Program Files`, where GMS is installed.
When I change something there, the snippets show up, but they have no hotkeys and the Preferences window hosts a hardcoded list of snippets, which does not adapt to the snippets files' contents.

Anyways, this way the snippets show up, so there must be something wrong with user snippets - maybe I missed something in the patch notes?

### Part 2
The code, the snippets generate, is not correct, they add blank characters and destroy the formatting after a `$` character, by indenting one additional char.
Please remove the inserted blank, if a snippet looks like this:
```
State-Enter:.add_state("|name|",$	function(sdata, prev_state) {$		$	}$)$ 
State-Step:,$function(sdata, frame) {$	$}$
State-Leave:,$function(sdata, new_state) {$	$}$
```
The problem is the `$function` -- the snippet shall put a comma at the end of the line, jump to the next line ($), and then create another function header. But this header is always indent by 1 blank character.
That has the effect, that, when using this snippet several times in a row, the code starts drifting more and more to the right and it's all but correct formatted.

### Part 3 - You can select "multiple" snippets at the same time
If you open the snippets popup, select an item with the cursor keys (but NOT press enter) and ALSO move the mouse on the popup, hovering over _another_ item (and NOT press a button), then, when you hit ENTER, both snippets are somehow generated at the same time, in a random order

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/19487451/4f487290-6e83-4a2b-89ce-93b0bc4f0a14)

leads to

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/19487451/5c276a8a-1135-48ff-9d5f-7e697c341ccc)

Thanks for reading
Cheers, Gris

### Steps To Reproduce
* Create a custom snippets.txt
* restart GMS
* do your snippets show up?

### Which version of GameMaker are you reporting this issue for?
IDE v2024.2.0.132 Runtime v2024.2.0.163

### Which operating system(s) are you seeing the problem on?
Windows 10.0.19045.0

[5643b77f-3349-49cf-981f-390e143fb9e9](https://api.gamemaker.io/api/github/downloads/5643b77f-3349-49cf-981f-390e143fb9e9)

**Comment by SmithTom6304** [comment]
Part 1 fixed - we were incorrectly skipping hotkeys that contained a hyphen "-" in their name, as we were being over-zealous when checking for old-style character hotkey bindings. This has now been fixed.

Part 2 I've not been able to reproduce - the function header seems to correctly be placed at the start of line. All functionality seems to match what is stated in the [code snippets manual page](https://manual.gamemaker.io/monthly/en/The_Asset_Editors/Code_Editor_Properties/Code_Snippets.htm). If QA could verify this, ideally on a Windows machine as I've only tested on mac.

Part 3 fixed - we were not handling our mouse over state correctly when switching between keyboard and mouse input. This has now been fixed.

Thanks for the report!

**Comment by Grisgram** [comment]
> Part 2 I've not been able to reproduce - the function header seems to correctly be placed at the start of line. All functionality seems to match what is stated in the [code snippets manual page](https://manual.gamemaker.io/monthly/en/The_Asset_Editors/Code_Editor_Properties/Code_Snippets.htm). If QA could verify this, ideally on a Windows machine as I've only tested on mac.

Please look at the history of my post - originally I had an image in there that show the indentation, it got removed...
I try to recover it

**Comment by Grisgram** [comment]
Assign those snippets from my opening post to your snippets.txt:

```
State-Enter:.add_state("|name|",$	function(sdata, prev_state) {$		$	}$)$ 
State-Step:,$function(sdata, frame) {$	$}$
State-Leave:,$function(sdata, new_state) {$	$}$
```

Then, in a code window, start with one "state-enter" snippet and where the cursor lands, repeat "state-step" as often as you like, you will see indent by 1 blank with every repeat, even when there is no blank charactor before the $ in the snippet

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/19487451/233a5690-8841-49dc-acd0-6df87f9b673b)

**Comment by SmithTom6304** [comment]
Issue with the function indenting should now be fixed. Function name should now correctly match up with the preceeding close bracket.

**Comment by Grisgram** [comment]
You're the best!!

Muchas gracias!

**Comment by alicemoretti** [comment]
All issues above have been verified fixed on IDE v2024.400.0.541  Runtime v2024.400.0.560. 

Thank you.
